apps/choleskey/choleskey_stdpar.cpp

```diff
@@ -0,0 +1,69 @@
+// Cholesky Decomposition: stdpar
+#include "argparse/argparse.hpp"
+#include "commons.hpp"
+
+#include <experimental/mdspan>
+#include <execution> // Include the execution header
+#include <vector>
+
+using namespace std;
+
+template<typename T>
+void Cholesky_Decomposition(std::vector<T> & vec, int n)
+{
+    std::vector<std::vector<T>> lower(n, std::vector<T>(n, 0));
+
+    using view_2d = std::extents<T, std::dynamic_extent, std::dynamic_extent>;
+    auto matrix = std::mdspan<T, view_2d, std::layout_right>(vec.data(), n,n);
+ 
+    // Decomposing a matrix into Lower Triangular
+    for (int i = 0; i < matrix.extent(0); i++) {
+        for (int j = 0; j <= i; j++) {
+            int sum = 0;
+ 
+            if (j == i) // summation for diagonals
+            {
+                std::for_each_n(std::execution::par, counting_iterator(0), j,
+                      [=, j = j, &sum](int32_t k) {
```

Comment:
for `-DSTDPAR=gpu` case, I wonder can we successfully pass `sum` variable to lambda function by reference? Will we encounter memory access error since `sum` has not allocated on GPU?

Reply:
I have tried to pass `sum` by value to avoid the memory warning, but got error issue:

> error: expression must be a modifiable lvalu
 sum += pow(lower[j][k], 2);
   ^

Reply:
how about using `transform_reduce` if this works for your purpose. 
```
                sum = std::transform_reduce(std::execution::par_unseq, lower[j].cbegin(), lower[j].cend(), 
                    sum, std::plus{},
                        [=, j = j](auto k) { return pow(lower[j][k], 2); });
```

Reply:
this perfectly resolved the capture issue, thanks